### PR TITLE
feat: graphql over websockets

### PR DESCRIPTION
**Pull request by tgriesser**
Closes UNIFY-974

Screen Shot 2022-02-09 at 10 51 39 PM

Adds a basic implementation of GraphQL over WS by shimming the `fetch` implementation, mainly so it doesn't show up prominently in the devtools console or in the command log while users' tests are executing. Currently only enabled in the `app` since that's where it's most prominent, though we could look into disabling in the launchpad if we want to be consistent.

Additionally:
- Fixes the prefetching of data from the `/__launchpad/preload` endpoint. 
- Changes to `cache-first` as the default cache strategy, which should help with race conditions and situations where mutation payloads aren't properly merged in the urql store (still need to investigate this one to better understand why this is happening)
- Adds `cy.withRetryableCtx` for assertions we want to run on the backend which might race against a user action that has just been taken, such as clicking a url issuing a mutation
- Adds `cy.__incorrectlyVisitAppWithIntercept` for situations where we're using `cy.intercept` in the app. We shouldn't be using this in the open mode tests, instead we should be stubbing/spying on the server and then asserting on value changes there.
- Adds `cy.tabUntil`, as a helper to call `cy.tab` until a condition matches - the current `cy.tab().tab()` was not helpful in situations where 

Added a menu item to toggle it back to fetch for debugging: 

image

### CODE REVIEW

File: packages/launchpad/cypress/e2e/migration.cy.ts [hunk 1]
```diff
@@ -67,7 +67,7 @@ function renameSupport (lang: 'js' | 'ts' = 'js') {
   }, { lang })
 }
 
-describe('Full migration flow for each project', () => {
+describe('Full migration flow for each project', { retries: { openMode: 2, runMode: 2 } }, () => {
```
Comment (lmiller1990) [inline packages/launchpad/cypress/e2e/migration.cy.ts:70]: Is this one really flaky?

Comment (tgriesser) [reply packages/launchpad/cypress/e2e/migration.cy.ts:70]: Yeah as far as I could tell

**Review by estrada9166 — DISMISSED** [review]

File: packages/frontend-shared/src/graphql/urqlClient.ts [hunk 2]
```diff
@@ -44,8 +54,8 @@ export async function preloadLaunchpadData () {
   try {
     const resp = await fetch('/__launchpad/preload')
 
-    window.__CYPRESS_INITIAL_DATA__ = JSON.parse(decodeBase64Unicode(await resp.json()))
-  } catch {
+    window.__CYPRESS_INITIAL_DATA__ = await resp.json()
+  } catch (e) {
```
[the remark this answers is not in the document]
Comment (tgriesser) [reply packages/frontend-shared/src/graphql/urqlClient.ts:58]: I guess log it? I'm not even this even needs to be in a `try/catch`, it shouldn't ever fail

File: packages/launchpad/cypress/e2e/config-files-error-handling.cy.ts [hunk 3]
```diff
@@ -60,6 +60,7 @@ describe('Config files error handling', () => {
     cy.get('body').should('not.contain.text', 'Cypress no longer supports')
   })
 
+  // NOTE: TODO: (tim) re-enable when we land the error fixes
```
Comment (ZachJW34) [inline packages/launchpad/cypress/e2e/config-files-error-handling.cy.ts:63]: What is this comment relate to? I don't see a skip or anything

**Review by ZachJW34 — APPROVED** [review]

**Review by ZachJW34 — APPROVED** [review]

File: packages/launchpad/cypress/e2e/config-files-error-handling.cy.ts (hunk 3 above)
Comment (tgriesser) [reply packages/launchpad/cypress/e2e/config-files-error-handling.cy.ts:63]: Oh, good call, I think it was failing at one point and I got it passing

**Review by ZachJW34 — DISMISSED** [review]

**Pull request merged**